.github/workflows/podvm_mkosi.yaml

```diff
@@ -97,32 +216,91 @@ jobs:
           remove-docker-images: "true"
 
       - name: Second checkout
+        if: ${{ inputs.arch == 'amd64' }}
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: "${{ inputs.git_ref }}"
 
-      - name: Decide on image to use
-        id: binaries-image
+
+      - name: Rebase the code
+        if: github.event_name == 'pull_request_target'
+        working-directory: ./
         run: |
-          if [ -z "${{ github.event.inputs.binaries-image }}" ]; then
-            echo "image=ghcr.io/${{ github.repository }}/podvm/binaries-fedora:${{ github.sha }}" | tee -a "$GITHUB_OUTPUT"
-          else
-            echo "image=${{ github.event.inputs.binaries-image }}" | tee -a "$GITHUB_OUTPUT"
-          fi
+          ./hack/ci-helper.sh rebase-atop-of-the-latest-target-branch
 
       - name: Download binaries and unpack into binaries-tree
         run: |
           crane export \
-            ${{ steps.binaries-image.outputs.image }} \
+            ${{ steps.build_binaries.outputs.image }} \
             podvm-binaries.tar
           mkdir -p podvm-mkosi/resources/binaries-tree
           tar xf podvm-binaries.tar -C podvm-mkosi/resources/binaries-tree
           rm podvm-binaries.tar
 
-      - name: Build image
+      - name: Build mkosi debug image
```

Comment:
if it's toggleable, we probably shouldn't call it debug image by default (build mkosi image) or maybe we can 2 steps with "if" clauses

Reply:
Thanks, I've done the later as I think it's a bit clearer even if there is a little bit of duplication in the workflow now.